engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/ConfluentSchemaRegistryClient.scala

```diff
@@ -1,14 +1,28 @@
 package pl.touk.nussknacker.engine.avro.schemaregistry.confluent
 
-import io.confluent.kafka.schemaregistry.client.{SchemaRegistryClient => ConfluenticSchemaRegistryClient}
+import cats.data.Validated
+import cats.data.Validated.{invalid, valid}
+import com.typesafe.scalalogging.LazyLogging
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException
+import io.confluent.kafka.schemaregistry.client.{SchemaRegistryClient => CSchemaRegistryClient}
 import org.apache.avro.Schema
-import pl.touk.nussknacker.engine.avro.AvroUtils
-import pl.touk.nussknacker.engine.avro.schemaregistry.SchemaRegistryClient
+import pl.touk.nussknacker.engine.avro.schemaregistry._
 
-trait ConfluentSchemaRegistryClient extends SchemaRegistryClient with ConfluenticSchemaRegistryClient {
-  override def getLatestSchema(subject: String): Schema =
-    AvroUtils.createSchema(getLatestSchemaMetadata(subject).getSchema)
+trait ConfluentSchemaRegistryClient extends SchemaRegistryClient with CSchemaRegistryClient with LazyLogging {
+  //Codes from https://docs.confluent.io/current/schema-registry/develop/api.html
+  protected val subjectNotFoundCode = 40401
+  protected val versionNotFoundCode = 40402
 
-  override def getBySubjectAndVersion(subject: String, version: Int): Schema =
-    AvroUtils.createSchema(getSchemaMetadata(subject, version).getSchema)
+  protected def handleClientError(schema: => Schema): Validated[SchemaRegistryError, Schema] =
+    try {
+      valid(schema)
+    } catch {
+      case exc: RestClientException if exc.getErrorCode == subjectNotFoundCode =>
+        invalid(InvalidSchema("Topic schema doesn't exist."))
```

Comment:
Based on message this situation not indicate that schema is invalid, but subject not found? also below maybe will be better `SchemaVersionNotFound`?

Reply:
Hmmm, right.